client/pd_service_discovery.go

```diff
@@ -174,7 +187,9 @@ func (c *pdServiceDiscovery) Init() error {
 	}
 	log.Info("[pd] init cluster id", zap.Uint64("cluster-id", c.clusterID))
 
-	c.updateServiceMode()
+	if err := c.checkServiceModeChanged(); err != nil {
+		log.Warn("[pd] failed to check service mode. will check later", zap.Error(err))
```

Comment:
```suggestion
		log.Warn("[pd] failed to check service mode, will check later", zap.Error(err))
```

Reply:
changed to "[pd] failed to check service mode and will check later"